### PR TITLE
Refactor with error fix

### PR DESCRIPTION
**Pull request by jhthorsen**
I've deprecated the "on_xxx" handlers for good in this commit.

I've fixed handling of ```$c->param('error')``` when used with ```$c->delay```.

I've removed blocking.t, since it's outdated.

I've refactored the code (sorry about that)

### CODE REVIEW

File: lib/Mojolicious/Plugin/OAuth2.pm [hunk 1]
```diff
@@ -49,98 +46,106 @@ sub register {
...
-          if (my $res = $tx->success) {
-            my $token = $self->_get_auth_token($res);
-            $args{callback}->($token) if $args{callback};
-            return $token;
-          }
-          elsif ($args{error_handler}) {
-            $args{error_handler}->($tx);
-          }
-        }
+        return $self->_got_code($c, @_);
```
Comment (marcusramberg) [inline lib/Mojolicious/Plugin/OAuth2.pm]: I don't really like _got_code/_got_error - How about _handle_code / _handle_error or _process_code / _process_error?

**Comment by marcusramberg** [comment]
LGTM apart from the comment - Mostly I don't like the passive form for method names.

**Comment by jhthorsen** [comment]
Renamed the method names. Merging it now.

**Comment by marcusramberg** [comment]
:sparkles: :fireworks: :hamster: :cat: :dog: :fireworks: :sparkles: 

Thanks for your contribution.

**Pull request merged**